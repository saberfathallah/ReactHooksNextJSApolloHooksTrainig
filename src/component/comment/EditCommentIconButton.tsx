import React from "react";

import IconButton from "@material-ui/core/IconButton";
import EditIcon from "@material-ui/icons/Edit";

interface EditCommentIconButton {
  setIsClickEdit: (boolean) => void;
}

const EditCommentIconButton = (props: EditCommentIconButton) => (
  <IconButton
    type="submit"
    onClick={(): void => props.setIsClickEdit(true)}
    aria-label="modify"
  >
    <EditIcon />
  </IconButton>
);

export default EditCommentIconButton;
