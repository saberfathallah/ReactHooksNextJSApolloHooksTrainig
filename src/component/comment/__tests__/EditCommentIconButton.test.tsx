import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import EditCommentIconButton from "../EditCommentIconButton";

describe("EditCommentIconButton", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<EditCommentIconButton setIsClickEdit={() => {}} />);
    expect(tree).toMatchSnapshot();
  });

  it("shouldn't call addLike should call dislike", () => {
    const setIsClickEdit = jest.fn();
    const wrapper = mount(<EditCommentIconButton setIsClickEdit={setIsClickEdit} />);
    wrapper.find("button").simulate("click");
    expect(setIsClickEdit).toHaveBeenCalledTimes(1);
  });
});
