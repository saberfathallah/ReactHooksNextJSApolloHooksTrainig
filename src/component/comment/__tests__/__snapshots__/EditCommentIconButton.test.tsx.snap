// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCommentIconButton should match snapshot 1`] = `
<button
  aria-label="modify"
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="submit"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
      />
    </svg>
  </span>
</button>
`;
